SteamKit2/SteamKit2/Steam/Handlers/SteamUnifiedMessages/SteamUnifiedMessages.cs

```diff
@@ -150,11 +191,29 @@ public AsyncJob<ServiceMethodResponse> SendMessage<TRequest>( string name, TRequ
         /// </summary>
         /// <typeparam name="TService">The type of a service interface.</typeparam>
         /// <returns>The <see cref="UnifiedService&lt;TService&gt;"/> wrapper.</returns>
-        public UnifiedService<TService> CreateService<TService>()
+        public UnifiedService<TService> CreateService<[DynamicallyAccessedMembers( Trimming.ForProtobufNet )] TService>()
         {
+            RegisterService<TService>();
             return new UnifiedService<TService>( this );
         }
 
+        /// <summary>
+        /// Registers a service type in order to recieve notifications. It is not neccesary to call this if
+        /// <see cref="CreateService{TService}"/> has been called with the same TService parameter.
+        /// </summary>
+        /// <typeparam name="TService">The type of a service interface.</typeparam>
+        public void RegisterService<[DynamicallyAccessedMembers( Trimming.ForProtobufNet )] TService>()
```

Comment:
Does this actually help with trimming?

Reply:
I think it should but I never got it working end to end trimmed anyway, never quite got past logon.